File: leetcode/algorithms/762_prime_number_of_set_bits_in_binary_representation/main.java
import java.util.stream.IntStream;

class PrimeNumberOfSetBitsInBinaryRepresentation {
    private boolean isPrime(int n) {
        if (n < 2) {
            return false;
        }

        for (int i = 2; i < n; i++) {
            if (n % i == 0) {
                return false;
            }
        }

        return true;
    }

    public int countPrimeSetBits(int left, int right) {
        int count = 0;

        for (int i = left; i <= right; i++) {
            int bits = 0;
            int n = i;

            while (n > 0) {
                bits += n % 2;
                n /= 2;
            }

            if (isPrime(bits)) {
                count++;
            }
        }

        return count;
    }


    // Solution
    // Solution 1
    public int solution1(int left, int right) {
        int count = 0;
        while (left <= right) {
            count += 665772 >> Integer.bitCount(left++) & 1;
        }
        return count;
    }

    // Solution 2: Stream
    public int solution2(int left, int right) {
        return IntStream.range(left, right + 1).map(i -> 665772 >> Integer.bitCount(i) & 1).sum();
    }
}

File: leetcode/algorithms/762_prime_number_of_set_bits_in_binary_representation/main.py
class PrimeNumberOfSetBitsInBinaryRepresentation:
    def isPrime(self, n: int) -> bool:
        if n < 2:
            return False

        for i in range(2, n):
            if n % i == 0:
                return False

        return True

    def countPrimeSetBits(self, left: int, right: int) -> int:
        count: int = 0

        for i in range(left, right + 1):
            bits: int = 0
            n: int = i

            while n > 0:
                bits += n % 2
                n //= 2

            if self.isPrime(bits):
                count += 1

        return count


    # Solution
    def solution(self, left: int, right: int) -> int:
        return sum(665772 >> bin(i).count("1") & 1 for i in range(left, right + 1))
